homeassistant/components/glances/sensor.py

```diff
@@ -354,110 +352,110 @@ def native_value(self) -> StateType:  # noqa: C901
         elif self.entity_description.key == "processor_load":
             # Windows systems don't provide load details
             try:
-                self._attr_native_value = value["load"]["min15"]
+                state = value["load"]["min15"]
             except KeyError:
-                self._attr_native_value = value["cpu"]["total"]
+                state = value["cpu"]["total"]
         elif self.entity_description.key == "process_running":
-            self._attr_native_value = value["processcount"]["running"]
+            state = value["processcount"]["running"]
         elif self.entity_description.key == "process_total":
-            self._attr_native_value = value["processcount"]["total"]
+            state = value["processcount"]["total"]
         elif self.entity_description.key == "process_thread":
-            self._attr_native_value = value["processcount"]["thread"]
+            state = value["processcount"]["thread"]
         elif self.entity_description.key == "process_sleeping":
-            self._attr_native_value = value["processcount"]["sleeping"]
+            state = value["processcount"]["sleeping"]
         elif self.entity_description.key == "cpu_use_percent":
-            self._attr_native_value = value["quicklook"]["cpu"]
+            state = value["quicklook"]["cpu"]
         elif self.entity_description.key == "docker_active":
             count = 0
             try:
                 for container in value["docker"]["containers"]:
                     if container["Status"] == "running" or "Up" in container["Status"]:
                         count += 1
-                self._attr_native_value = count
+                state = count
             except KeyError:
-                self._attr_native_value = count
+                state = count
         elif self.entity_description.key == "docker_cpu_use":
             cpu_use = 0.0
             try:
                 for container in value["docker"]["containers"]:
                     if container["Status"] == "running" or "Up" in container["Status"]:
                         cpu_use += container["cpu"]["total"]
-                    self._attr_native_value = round(cpu_use, 1)
+                    state = round(cpu_use, 1)
             except KeyError:
-                self._attr_native_value = STATE_UNAVAILABLE
+                state = STATE_UNAVAILABLE
         elif self.entity_description.key == "docker_memory_use":
             mem_use = 0.0
             try:
                 for container in value["docker"]["containers"]:
                     if container["Status"] == "running" or "Up" in container["Status"]:
                         mem_use += container["memory"]["usage"]
-                    self._attr_native_value = round(mem_use / 1024**2, 1)
+                    state = round(mem_use / 1024**2, 1)
             except KeyError:
-                self._attr_native_value = STATE_UNAVAILABLE
+                state = STATE_UNAVAILABLE
```

Comment:
We don't set `STATE_UNAVAIBLE` directly. We should use the `available` property or attribute for that.

We should consider if this is the correct state for this case though. Unknown values should be represented with unknown state by setting `None` as local state which will be converted by the base Entity class to `STATE_UNKNOWN`. What does a KeyError mean?

Reply:
I plan to do a followup PR that simplifies the sensor state update. I have already added a method for reformatting the data collected by the glances_api library (https://github.com/home-assistant-ecosystem/python-glances-api/pull/11). This will allow us to add only the available sensors (so if there are no containers, the "docker" sensor won't be added).
The current code adds all sensors and the keyerror is used to catch in case the sensor data doesn't exist.

Reply:
Ok, let's follow up later for this one then.